plugins/woocommerce/includes/class-wc-emails.php

```diff
@@ -217,20 +225,36 @@ public function __construct() {
 	public function init() {
 		// Include email classes.
 		include_once dirname( __FILE__ ) . '/emails/class-wc-email.php';
+		$this->default_emails_paths = $this->default_emails_paths();
+       	$emails_data = apply_filters( 'woocommerce_before_email_classes', $this->default_emails_paths );
```

Comment:
While I totally get why the filter has this name I think it'd be best to use a different one, as before/after hooks are usually actions (not filters). I suggest we use `woocommerce_default_emails` (or similar) and apply it directly inside the (suggested) method `get_default_emails()` instead of up here.

That way, any modification to the default classes would be reflected when you call `get_default_emails()`, and not only during initialization.

Also, docblocks are required [for new hooks](https://github.com/woocommerce/woocommerce/wiki/Adding-Actions-and-Filters).